src/zenml/zen_server/utils.py

```diff
@@ -29,7 +29,7 @@
 )
 from urllib.parse import urlparse
 
-import secure
+# import secure
```

Comment:
Why was this import moved inside the function?

Reply:
The `zenml.zen_server.utils` gets imported from numerous modules, whereas the `secure` package is only used by a function that is used by the `[server]` extra. Moving this inside the function allows me to move the `secure` dependency to the extra.

Reply:
I'm still not sure about this. `zenml.zen_server.utils` should only be used in the server, so it should assume that the server extra is installed. If that is not the case, we should instead fix our endpoints IMO

Reply:
> zenml.zen_server.utils should only be used in the server.

Overall, I agree with this statement. However, it is used by quite a lot of files at this time. While a significant chunk of them are endpoints, the rest consist mostly of some type-checking stuff, some CLI and`plugin_registry` code. Instead of checking whether a rework would be necessary on all of these files, moving the import inside the function (and moving the dependency) seemed like a more efficient solution here unless you see a problem that it would create due to this move.

Reply:
It does not create a problem, it's still the wrong solution IMO. `Any` returns are always bad for type checking, and in this case we could avoid them quite easy I think: Just moving the two secure header functions into a separate file, which will then only get imported by the server.

Reply:
I liked the idea of separating them into a different file 👍  Implemented it, feel free to check.